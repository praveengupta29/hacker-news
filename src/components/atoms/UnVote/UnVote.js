import React from 'react';
import PropTypes from 'prop-types';

import Anchor from '../Anchor';
import './UnVote.scss';

const UnVote = ({ voteFeed, objectId }) => {
  return (
    <Anchor
      handleLinkClick={e => {
        e.preventDefault();
        voteFeed({
          objectId,
          voted: false,
        });
      }}
      className="un-vote-link secondary-color"
    >
      unvote
    </Anchor>
  );
};

UnVote.propTypes = {
  objectId: PropTypes.string.isRequired,
  voteFeed: PropTypes.func.isRequired,
};

export default UnVote;
