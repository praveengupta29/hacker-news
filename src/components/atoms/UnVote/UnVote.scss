.un-vote-link {
  margin-right: 4px;
}
