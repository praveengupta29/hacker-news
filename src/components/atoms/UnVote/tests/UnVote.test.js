import React from 'react';
import { shallow } from 'enzyme';
import UnVote from '../UnVote';

describe('UnVote component', () => {
  const props = {
    voteFeed: () => {},
    objectId: '6382672',
  };

  it('should render correctly and match snapshot', () => {
    const component = shallow(<UnVote {...props} />);
    expect(component).toMatchSnapshot();
  });
});
