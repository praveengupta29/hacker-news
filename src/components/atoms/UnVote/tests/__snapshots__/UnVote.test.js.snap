// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnVote component should render correctly and match snapshot 1`] = `
<Anchor
  className="un-vote-link secondary-color"
  handleLinkClick={[Function]}
  target="_self"
  to=""
>
  unvote
</Anchor>
`;
